File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
let traverse = (node, traversal) => {
    if(node === null)
        return
    traverse(node.left, traversal)
    traverse(node.right, traversal)
    traversal.push(node.val)
}


var postorderTraversal = function(root) {
    let postorderTraversal = []
    traverse(root, postorderTraversal)
    
    return postorderTraversal
    
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    public List<Integer> traversal(TreeNode node, List<Integer> traversal) {
        if(node != null) {
            traversal(node.left, traversal);
            traversal(node.right, traversal);
            traversal.add(node.val);
        }
        
        return traversal;
    }
    
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> traversal = new ArrayList<>();
        
        return traversal(root, traversal);
    }
}